### PR TITLE
Reorganize dependencies in axlsx.rb

### PR DESCRIPTION
**Pull request by tagliala**
Requires #398

---

Reorganize dependencies in axlsx.rb 

- Group external and core gem dependencies at the top of the file
- Ensure all dependencies are loaded before internal library files

This change improves the organization and readability of the axlsx.rb
file, making it easier for developers to understand and maintain the
library's dependency structure. The new arrangement provides a clearer
overview of both external and core dependencies required by the axlsx
library.

### CODE REVIEW

File: lib/axlsx.rb [hunk 1]
```diff
@@ -1,8 +1,19 @@
 # frozen_string_literal: true
 
-require 'htmlentities'
 require 'axlsx/version'
+
+# Required dependencies
```
Comment (kiskoza) [inline lib/axlsx.rb:5]: can we call it `gem dependencies`, `gemspec dependencies` or something similar? all of the non-optional dependencies are required dependencies and we're requiring them in the next lines - this word might be overused in this context

File: lib/axlsx.rb [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+require 'htmlentities'
 require 'marcel'
+require 'nokogiri'
+require 'zip'
+
+# Core dependencies
```
Comment (kiskoza) [inline lib/axlsx.rb:11]: can we extend it to be `Ruby core depencencies`?

**Comment by tagliala** [comment]
Done but please wait, I think there is something wrong about bigdecimal

**Comment by tagliala** [comment]
Ref: jnunemaker/crack#79 😅

**Comment by tagliala** [comment]
Tracked in #404

**Pull request merged**